### PR TITLE
Reduce some allocations with the magic of spans etc.

### PR DESCRIPTION
**Pull request by cvium**
**Changes**
euh

### CODE REVIEW

**Review by barronpm — APPROVED** [review]

File: MediaBrowser.Common/Extensions/SplitLinesStringExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        /// Creates a new line split enumerator.
+        /// </summary>
+        /// <param name="str">The span to split.</param>
+        /// <param name="separator">The separator to split on.</param>
+        /// <returns>The enumerator struct.</returns>
+        [Pure]
+        public static LineSplitEnumerator Split(this ReadOnlySpan<char> str, char separator) => new (str, separator);
+
+        [StructLayout(LayoutKind.Auto)]
+        public ref struct LineSplitEnumerator
```
Comment (Bond-009) [inline MediaBrowser.Common/Extensions/SplitLinesStringExtensions.cs:57]: This isn't a `LineSplitEnumerator` anymore

**Review by Bond-009 — APPROVED** [review]

File: MediaBrowser.Common/Extensions/SplitLinesStringExtensions.cs (hunk 1 above)
Comment (cvium) [reply MediaBrowser.Common/Extensions/SplitLinesStringExtensions.cs:57]: True. I'm terrible with names though

Comment (Bond-009) [reply MediaBrowser.Common/Extensions/SplitLinesStringExtensions.cs:57]: Just `SplitEnumerator`?

File: MediaBrowser.Common/Extensions/SplitStringExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+                var span = _str;
+                var index = span.IndexOf(_separator);
+                if (index == -1)
+                {
+                    _str = ReadOnlySpan<char>.Empty;
+                    Current = span;
+                    return true;
+                }
+
+                if (index < span.Length - 1 && span[index] == _separator)
```
Comment (Bond-009) [inline MediaBrowser.Common/Extensions/SplitStringExtensions.cs:89]: This block doesn't do anything?

File: tests/Jellyfin.Server.Implementations.Tests/Data/SqliteItemRepositoryTests.cs [hunk 3]
```diff
@@ -76,7 +96,7 @@ public static IEnumerable<object[]> ItemImageInfoFromValueString_Valid_TestData(
         [MemberData(nameof(ItemImageInfoFromValueString_Valid_TestData))]
         public void ItemImageInfoFromValueString_Valid_Success(string value, ItemImageInfo expected)
         {
-            var result = _sqliteItemRepository.ItemImageInfoFromValueString(value);
+            var result = _sqliteItemRepository.ItemImageInfoFromValueString(value.AsSpan());
```
Comment (Bond-009) [inline tests/Jellyfin.Server.Implementations.Tests/Data/SqliteItemRepositoryTests.cs:99]: The `AsSpan` shouldn't be needed, right?

**Review by Bond-009 — CHANGES_REQUESTED** [review]

**Review by Bond-009 — APPROVED** [review]

**Pull request merged**